src/main/java/io/javalin/core/JavalinServlet.kt

```diff
@@ -74,7 +77,25 @@ class JavalinServlet(
                 }
             }
             if (endpointEntries.isEmpty() && type != HandlerType.HEAD && type != HandlerType.GET) {
-                throw HaltException(404, "Not found")
+
+                if (prefer405over404) {
+                    val availableHandlerTypes = MethodNotAllowedUtil.findAvailableHandlerTypes(matcher, requestUri)
```

Comment:
Calling this here will do matcher work twice: first for called `HandlerType`, then second time here.
Consider searching for all the handlers matching uri in the beginning of the method and then reusing that pool for here.

Reply:
The performance penalty is only applicable for 404s (well, 405s), so it doesn't really matter. I prefer doing the work twice since it keeps the functionality isolated.